File: csharp/reverse-string/ReverseString.cs
using System;

public static class ReverseString
{
    public static string Reverse(string input)
    {
        var output = new char[input.Length];

        for (var i = 0; i < input.Length; ++i)
            output[output.Length - 1 - i] = input[i];

        return String.Join("", output);
    }
}

File: java/reverse-string/src/main/java/ReverseString.java
class ReverseString {
    String reverse(String inputString) {
        var size = inputString.length();
        var output = new char[size];

        for (var i = 0; i < size; ++i)
            output[i] = inputString.charAt(size - i - 1);

        return new String(output);
    }
}
